src/aws-cpp-sdk-core/include/aws/core/utils/crypto/PrecalculatedHash.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#pragma once
+
+#include <aws/core/Core_EXPORTS.h>
+#include <aws/core/utils/crypto/Hash.h>
+#include <aws/core/utils/crypto/HashResult.h>
+#include "aws/core/utils/Outcome.h"
+
+namespace Aws {
+    namespace Utils {
+        namespace Crypto {
+            class AWS_CORE_API PrecalculatedHash : public Hash {
```

Comment:
please add docstring comments to this class 
and why it is needed, i.e. what is the difference with a regular hash?